unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+package org.unicode.text.UCD;
+
+import com.ibm.icu.text.UnicodeSet;
+import com.ibm.icu.util.VersionInfo;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.unicode.props.IndexUnicodeProperties;
+import org.unicode.props.UcdProperty;
+import org.unicode.props.UcdPropertyValues;
+import org.unicode.props.UnicodeProperty;
+import org.unicode.props.UnicodePropertySymbolTable;
+import org.unicode.text.utility.Settings;
+
+/**
+ * This class implements the semantics of property-query as defined in the UnicodeSet specification.
+ */
+public class VersionedSymbolTable extends UnicodeSet.XSymbolTable {
+    private VersionedSymbolTable() {}
+
+    public static VersionedSymbolTable forReview() {
+        var result = new VersionedSymbolTable();
+        result.requireSuffixForLatest = true;
+        result.implicitVersion = Settings.LAST_VERSION_INFO;
+        result.previousVersion = Settings.LAST2_VERSION_INFO;
+        return result;
+    }
+
+    public static VersionedSymbolTable forDevelopment() {
+        var result = new VersionedSymbolTable();
+        result.requireSuffixForLatest = false;
+        result.implicitVersion = Settings.LATEST_VERSION_INFO;
+        result.previousVersion = Settings.LAST_VERSION_INFO;
+        return result;
+    }
+
+    public VersionedSymbolTable setUnversionedExtensions(UnicodeProperty.Factory factory) {
+        unversionedExtensions = factory;
+        return this;
+    }
+
+    /**
+     * Parses a string prefixed with an optional-version-qualifier. If there is a version-qualifier,
+     * returns the corresponding VersionInfo and removes the prefix from the given StringBuilder.
+     */
```

Comment:
```suggestion
    /**
     * Parses a string prefixed with an optional-version-qualifier. If there is a version-qualifier,
     * returns the corresponding VersionInfo and removes the prefix from the given StringBuilder.
     * Otherwise returns null.
     */
```